packages/relay/src/lib/eth.ts

```diff
@@ -177,28 +185,50 @@ export class EthImpl implements Eth {
     this.logger.trace(`${requestIdPrefix} feeHistory(blockCount=${blockCount}, newestBlock=${newestBlock}, rewardPercentiles=${rewardPercentiles})`);
 
     try {
-      const latestBlockNumber = await this.translateBlockTag(EthImpl.blockLatest, requestId);
-      const newestBlockNumber = (newestBlock == EthImpl.blockLatest || newestBlock == EthImpl.blockPending)
-        ? latestBlockNumber
-        : await this.translateBlockTag(newestBlock, requestId);
-
-      if (newestBlockNumber > latestBlockNumber) {
-        return predefined.REQUEST_BEYOND_HEAD_BLOCK(newestBlockNumber, latestBlockNumber);
+      let newestBlockNumber;
+      let latestBlockNumber;
+      if(this.getEthFeeHistoryFixedFee()) {
+        newestBlockNumber = (newestBlock == EthImpl.blockLatest || newestBlock == EthImpl.blockPending)
+            ? await this.translateBlockTag(EthImpl.blockLatest, requestId)
+            : await this.translateBlockTag(newestBlock, requestId);
+      } else { // once we finish testing and refining Fixed Fee method, we can remove this else block to clean up code
+        latestBlockNumber = await this.translateBlockTag(EthImpl.blockLatest, requestId);
+        newestBlockNumber = (newestBlock == EthImpl.blockLatest || newestBlock == EthImpl.blockPending)
+            ? latestBlockNumber
+            : await this.translateBlockTag(newestBlock, requestId);
+
+        if (newestBlockNumber > latestBlockNumber) {
+          return predefined.REQUEST_BEYOND_HEAD_BLOCK(newestBlockNumber, latestBlockNumber);
+        }
       }
 
       blockCount = blockCount > maxResults ? maxResults : blockCount;
 
       if (blockCount <= 0) {
         return EthImpl.feeHistoryZeroBlockCountResponse;
       }
+      let feeHistory: object | undefined;
 
-      const cacheKey = `${constants.CACHE_KEY.FEE_HISTORY}_${blockCount}_${newestBlock}_${rewardPercentiles?.join('')}`;
-      let feeHistory: object | undefined = this.cache.get(cacheKey);
-      if (!feeHistory) {
-        feeHistory = await this.getFeeHistory(blockCount, newestBlockNumber, latestBlockNumber, rewardPercentiles, requestId);
-        if (newestBlock != EthImpl.blockLatest && newestBlock != EthImpl.blockPending) {
-          this.logger.trace(`${requestIdPrefix} caching ${cacheKey}:${JSON.stringify(feeHistory)} for ${constants.CACHE_TTL.ONE_HOUR} ms`);
-          this.cache.set(cacheKey, feeHistory);
+      if(this.getEthFeeHistoryFixedFee()) {
+
+        let oldestBlock = newestBlockNumber - blockCount + 1;
+        if(oldestBlock <= 0) {
```

Comment:
I feel like we should just error out in this case.
Also this will only happen on new envs and more likely local envs. Even then by the time users start hitting it i'm sure 10 blocks have been made.

Reply:
I leave it there cause there were 1 Unit Test that had the test case.

Reply:
Ahh this is the equivalent of `const oldestBlockNumber = Math.max(0, newestBlockNumber - blockCount + 1);`
I see. Okay.
Can reuse that code if you like

Reply:
Opened another [ticket](https://github.com/hashgraph/hedera-json-rpc-relay/issues/1099) for taking care of this in the future.